lightning/src/ln/channel.rs

```diff
@@ -6027,13 +6047,17 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 			let _dummy: u32 = Readable::read(reader)?;
 		}
 
-		let counterparty_forwarding_info = match <u8 as Readable>::read(reader)? {
-			0 => None,
-			1 => Some(CounterpartyForwardingInfo {
-				fee_base_msat: Readable::read(reader)?,
-				fee_proportional_millionths: Readable::read(reader)?,
-				cltv_expiry_delta: Readable::read(reader)?,
-			}),
+		// Read fields for `CounterpartyForwardingInfo`. As the `outbound_htlc_maximum_msat` is a
+		// TLV Optional variant, we construct the `CounterpartyForwardingInfo` struct after it ha
+		// been read.
+		let (has_forwarding_info, fee_base_msat, fee_proportional_millionths, cltv_expiry_delta) = match <u8 as Readable>::read(reader)? {
```

Comment:
This may be more readable if we just create a `CounterpartyForwardingInfo` object with the `htlc_maximum_msat` and `htlc_minimum_msat` fields initialized to some default.